### PR TITLE
Adding AllowAnonymousToPage & AllowAnonymousToFolder

### PR DESCRIPTION
**Pull request by hishamco**
After my last PR #5942, I notice that we missed some of auth functions after migrate the RazorPages code from the old repo https://github.com/aspnet/RazorPages/blob/master/src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/PageModelConventionBuilderExtensions.cs#L14-L24, also I see @pranavkm mentioned this too in the issue #5884, so I created this PR to bring them back

### CODE REVIEW

**Comment by pranavkm** [comment]
@ryanbrandenburg this takes care of the second bullet from https://github.com/aspnet/Mvc/issues/5884. Thanks @hishamco

**Comment by hishamco** [comment]
Is there anything blocking this?!!
⌚️

**Comment by pranavkm** [comment]
We should probably add a functional test for this - something along the lines of requiring auth for a directory, but allowing anonymous to the Login page under it. Just to make sure these features work when used in tandem.

**Comment by hishamco** [comment]
I 'll try to add a functional test for this ASAP

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 1]
```diff
@@ -249,6 +249,21 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
...
+            
+            // Act
+            var response = await Client.GetAsync(url);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var content = await response.Content.ReadAsStringAsync();
+            Assert.Equal("Login Page", content);
+        }
+
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Extra newline. Should be good check in once the CI builds complete.

Comment (hishamco) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: You know what I was add it, after that I delete it 😄 , coz I didn't proper guideline for `var` tests.

Sure I will add it once the CI builds complete

**Comment by pranavkm** [comment]
AppV is complaining about compilation errors. Could you have a look?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 2]
```diff
@@ -249,6 +249,22 @@ public RazorPagesTest(MvcTestFixture<RazorPagesWebSite.Startup> fixture)
             Assert.Equal("/Login?ReturnUrl=%2FHelloWorldWithAuth", response.Headers.Location.PathAndQuery);
         }
 
+        [Fact]
+        public async Task AuthorizePage_AllowAnonymousForSpecificPages()
+        {
+            // Arrange
+            var url = "/Admin/Login";
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Need to use `/Pages/Admin/Login`.

Comment (hishamco) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: @pranavkm don't worry I will take care about all of these, I'm already fixing them in my end. The main reason for that is after I download the latest bits, some packages are not restored probably, so I can't run the test in my end, hope I got a fix soon

thanks

**Comment by hishamco** [comment]
@pranavkm it's fine now

**Comment by pranavkm** [comment]
Nice. Thanks @hishamco

**Pull request merged**